tests/integration/components/au-button-test.js

```diff
@@ -23,4 +23,29 @@ module('Integration | Component | au-button', function (hooks) {
 
     assert.dom(this.element).hasText('template block text');
   });
+
+  test('it hides the text when @hideText is set to true', async function (assert) {
```

Comment:
Even though I appreciate it _a lot_ that you added tests 🎉 , I think these kind of tests aren't super useful in general 😅. Checking for certain classes isn't a good practice (imo) since it's an implementation detail and it doesn't guarantee that the applied css actually makes the text hidden.

We actually want to test if things are visually hidden, which we now can because of the Chromatic setup that Dieter added a while ago (#267).

I'm not 100% sure, but I think Chromatic makes screenshots based on the stories. So I think we can add a new story for the "Icon only" button and that way we will know for certain if something breaks the text hiding in the future.

Reply:
> Even though I appreciate it _a lot_ that you added tests 🎉 , I think these kind of tests aren't super useful in general 😅. Checking for certain classes isn't a good practice (imo) since it's an implementation detail and it doesn't guarantee that the applied css actually makes the text hidden.

Wasn't absolutely sure about that one too 🫣 I kept looking for an actual solution to check for visibility as `isVisible` only checks for the element dimensions, but couldn't find anything.

> We actually want to test if things are visually hidden, which we now can because of the Chromatic setup that Dieter added a while ago (#267).
> 
> I'm not 100% sure, but I think Chromatic makes screenshots based on the stories. So I think we can add a new story for the "Icon only" button and that way we will know for certain if something breaks the text hiding in the future.

This seems like a way better solution 👍 I'll give it a look when I have the time. Thanks for mentioning it!

@Windvis I'll delete this test, but is the last one I added worth keeping?

Reply:
Hmm, I'm not sure if I'm a fan of that class to be honest 😅. Making css classes part of the components "public API" of a component is not something I like personally, but I've seen it done before.

I would probably wrap the component in the project and add the class myself. Something like `<IconButton @icon="something">Hidden text</IconButton>` which would set the `@hideText` arg and the icon-only class you need for your projects styling, and passes through all the other arguments the button supports.

If it is actually considered public API then yes I think the test makes sense.

Reply:
> This seems like a way better solution 👍 I'll give it a look when I have the time. Thanks for mentioning it!

I _think_ it is as simple as adding a new [named export ](https://github.com/appuniversum/ember-appuniversum/blob/f8118bcfe5880e94a0aae9a0d2c62b361532cefa/stories/5-components/Buttons/AuButton.stories.js#L117-L129) with the proper arg values.

Reply:
> Hmm, I'm not sure if I'm a fan of that class to be honest 😅. Making css classes part of the components "public API" of a component is not something I like personally, but I've seen it done before.
> 
> I would probably wrap the component in the project and add the class myself. Something like `<IconButton @icon="something">Hidden text</IconButton>` which would set the `@hideText` arg and the icon-only class you need for your projects styling, and passes through all the other arguments the button supports.
> 
> If it is actually considered public API then yes I think the test makes sense.

I think I'll hand over the decision on that to @Dietr. Do you have a taken on this?

Reply:
As an other option, I could also make the visual changes for the icon-only button (so it has the same width as height) here inside `ember-appuniversum`, but I'm not sure this is a common request. As a plus, we could also size up the icon a bit as it is done inside `AuLink`. https://github.com/appuniversum/ember-appuniversum/blob/b73e76e8ff51713065c703e05194b2bfa08ee8a2/app/styles/ember-appuniversum/_c-link.scss#L81

Reply:
I did a quick check in our project and the icon inside the icon-only version of the button is really small at the moment.

(on the left: our own component - on the right: `AuButton`)
<img width="411" alt="CleanShot 2022-08-26 at 10 58 06@2x" src="https://user-images.githubusercontent.com/18174827/186867701-ca9d4253-4443-48dc-8bdb-9ff4a615f369.png">

Maybe I could kill two birds with one stone and:
- Adjust the icon size inside the icon-only button
- Adjust horizontal padding of the icon-only button here inside `ember-appuniversum` or use the `au-c-button--icon-only` modifier in our own project.

Thoughts?

Reply:
I found this comment by Dieter: https://github.com/lblod/ember-submission-form-fields/blob/46af3ccef7a0045546a02cdc0c4ca311fe30ba77/addon/components/rdf-input-fields/concept-scheme-multi-select-checkboxes.hbs#L20-L21

It seems Webuniversum does have an icon only variant, so if the Appuniversum version doesn't look like that I assume it makes sense to fix it here?

Reply:
So to check off ✅: the styling changes are ok, but remove the remaining test I added in favour of a visual regression test?

Reply:
That's what I would do, yea, but @Dietr might have a different vision?

Reply:
I'm fine with the changes you propose. A bigger icon in the icon only button seems like a good improvement for all projects 👍